dubbo-remoting/dubbo-remoting-zookeeper/src/main/java/org/apache/dubbo/remoting/zookeeper/support/AbstractZookeeperClient.java

```diff
@@ -45,6 +46,8 @@
 
     private volatile boolean closed = false;
 
+    private final Set<String>  persistentExistNodePath = Sets.newConcurrentHashSet();
+

```

Comment:
If a path is deleted, the path should be removed from the set.

Reply:
> If a path is deleted, the path should be removed from the set.
From me , I think we delete only the non persistent node. I will have a look with the code

Reply:
pls. avoid of using guava. It's introduced from